webassets_elm/__init__.py

```diff
@@ -30,16 +44,19 @@ def input(self, _in, out, **kw):
         the compiled contents to a temporary file and then read it in order to
         output to stdout.
         """
-
         # create a temp file
         tmp = NamedTemporaryFile(suffix='.js', delete=False)
         tmp.close()  # close it so windows can read it
 
         # write to a temp file
         elm_make = self.binary or 'elm-make'
+        change_directory = bool(self.change_directory or False)
         source = kw['source_path']
+        source_dir = os.path.join(*os.path.split(source)[:-1])
+        exec_dir = source_dir if change_directory else getcwd()
         write_args = [elm_make, source, '--output', tmp.name, '--yes']
-        with TemporaryFile(mode='w') as fake_write_obj:
+        with excursion(exec_dir), \
```

Comment:
Which ever way you want is fine with me. I tend to use multiple lines when there's more than one contextmanager because I find it easier to scan visually (and they tend to exceed 80 chars anyway). But you're right that it's not strictly necessary here.